Classes/Bookmark v2/BookmarkViewController.swift

```diff
@@ -0,0 +1,304 @@
+        let bookmark = viewModel.bookmark
+        let destinationViewController: UIViewController
+
+        switch bookmark.type {
+        case .repo:
+            let repo = RepositoryDetails(
+                owner: bookmark.owner,
+                name: bookmark.name,
+                defaultBranch: bookmark.defaultBranch,
+                hasIssuesEnabled: bookmark.hasIssueEnabled
+            )
+            destinationViewController = RepositoryViewController(client: client, repo: repo)
+
+        case .issue, .pullRequest:
+            let issueModel = IssueDetailsModel(
+                owner: bookmark.owner,
+                repo: bookmark.name,
+                number: bookmark.number
+            )
+            destinationViewController = IssuesViewController(client: client, model: issueModel)
+        default:
+            return
+        }
+        let navigation = UINavigationController(rootViewController: destinationViewController)
+        showDetailViewController(navigation, sender: nil)
+    }
+
+    func didDelete(bookmarkSectionController: BookmarkSectionController, viewModel: BookmarkViewModel) {
+        bookmarkStore.remove(viewModel.bookmark)
+        update(animated: true)
+    }
+
+    // MARK: - ListAdapterDataSource
+
+    func objects(for listAdapter: ListAdapter) -> [ListDiffable] {
+        var bookmarks: [ListDiffable]
+        switch state {
+        case .idle:
+            bookmarks = bookmarkStore.values.flatMap { BookmarkViewModel(bookmark: $0, width: view.bounds.width) }
+        case .filtering(let term):
+            bookmarks = filtered(array: bookmarkStore.values, query: term)
+                .flatMap { bookmark in
+                    BookmarkViewModel(bookmark: bookmark, width: view.bounds.width)
+            }
+        }
+
+        if !bookmarks.isEmpty {
+            bookmarks.insert(headerKey, at: 0)
+        }
+
+        return bookmarks
+    }
+
+    func listAdapter(_ listAdapter: ListAdapter, sectionControllerFor object: Any) -> ListSectionController {
+        guard let object = object as? ListDiffable else { fatalError() }
+
+        if object === headerKey {
+            return BookmarkHeaderSectionController(delegate: self)
+        } else if object is BookmarkViewModel {
+            return BookmarkSectionController(delegate: self)
+        }
+
+        fatalError("Could not find section controller for object")
+    }
+
+    func emptyView(for listAdapter: ListAdapter) -> UIView? {
+        guard bookmarkStore.values.isEmpty else { return nil }
+
+        let view = InitialEmptyView(
+            imageName: "bookmarks-large",
+            title: NSLocalizedString("Add Bookmarks", comment: ""),
+            description: NSLocalizedString("Bookmark your favorite issues,\npull requests, and repositories.", comment: "")
+        )
+        view.delegate = self
+        return view
+    }
+
+    // MARK: - UISearchBarDelegate
```

Comment:
nit: `// MARK: ` style inconsistent w/ rest of app. No "-"

Reply:
There's a SwiftLint rule to ensure marks all follow the same style. I think it chooses to include the `-` but it would enforce everyone to use same? Just an idea!